### DISCUSSION TITLE
RDKit (2023-03-02): Non-Standard InChI and InChIKey for Large Molecules

### DISCUSSION THREAD

**Discussion by Lili-Summer** [rdkit/rdkit#7553] [Q&A]
Hello RDKit community,

I'm currently working on generating non-standard InChI and InChIKey representations using RDKit. However, I'm encountering an issue where RDKit is reporting the error: Too many atoms [did you forget 'LargeMolecules' switch?]. I've reviewed the InChI options available but haven't found 'LargeMolecules' listed among them. I also attempted to include /FixedH/LargeMolecules in the options, but unfortunately, this didn't resolve the problem.

Attached is a snippet of the code I'm using to attempt generating these representations. Could someone please provide guidance or suggestions on how to address this issue?

Thank you for your assistance!

Best,
Leila
[ns_InChI_InChIkey.txt](https://github.com/user-attachments/files/15934316/ns_InChI_InChIkey.txt)

**Comment by Lili-Summer** [comment]
Hello RDKit community,

An update to my previous post:
- I found that the error is within the InChI library.
- It seems InChI v1.05 and later versions supports the InChI and InChIKey for large molecules (more than 1024 atoms). 
- In my case after adding H to calculate ns-InChI and ns-InChIKey, the number of atoms increases to 1063 and InChI complains about that. 
- Now my question is: Does RDKit 2023-03-02 supports InChI release 1.05 (I don't know how to get the InChI version).

Thanks!
Leila

**Comment by Melclic** [comment]
I had a similar issue and to solve it I needed to add the flag in the following format:

`-LargeMolecules `

Hope this helps

**Reply by JamesThompsonC** [reply]
Thank you Meclic. I'm not sure where that 'flag' goes? How to use it? Is it an argument in the Chem.inchi.MolToInchiKey() function?

**Reply by Melclic** [reply]
As so: 
`created_inchi_key = AllChem.MolToInchiKey(created_mol, options='-LargeMolecules')`
